pkg/build/controller/image_change_controller.go

```diff
@@ -110,22 +118,24 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageStream) erro
 				ObjectMeta: kapi.ObjectMeta{
 					Name: config.Name,
 				},
-				TriggeredByImage: triggeredImage,
+				TriggeredByImage: &kapi.ObjectReference{
+					Kind: "DockerImage",
+					Name: triggeredImage,
+				},
 			}
 			if _, err := c.BuildConfigInstantiator.Instantiate(config.Namespace, request); err != nil {
 				if kerrors.IsConflict(err) {
-					// This is not a retryable error. The worst case outcome of not updating the buildconfig
-					// is that we might rerun a build for the same "new" imageid change in the future,
-					// which is better than guaranteeing we run the build 2+ times by retrying it here.
-					return ImageChangeControllerFatalError{
-						Reason: fmt.Sprintf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err),
-						Err:    err,
-					}
+					util.HandleError(fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err))
+				} else {
+					util.HandleError(fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err))
 				}
-
-				return fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err)
+				hasError = true
+				continue
 			}
 		}
 	}
+	if hasError {
+		return errors.New(fmt.Sprintf("an error occurred processing 1 or more build configurations; the image change trigger for image stream %s/%s:%s will be retried", repo.Namespace, repo.Status.DockerImageRepository))
```

Comment:
Sorry, I originally wrote errors.New because the text had no varargs in it. But now it does...